### PR TITLE
clearCreds redirect behavior depends on request type.

### PR DESCRIPTION
**Pull request by StevenXL**
Before submitting your PR, check that you've:

- [x] Bumped the version number
- [x] Documented new APIs with [Haddock markup](https://www.haskell.org/haddock/doc/html/index.html)
- [x] Added [`@since` declarations](http://haskell-haddock.readthedocs.io/en/latest/markup.html#since) to the Haddocks for new, public APIs

After submitting your PR:

- [x] Update the Changelog.md file with a link to your PR
- [ ] Check that CI passes (or if it fails, for reasons unrelated to your change, like CI timeouts)

### CODE REVIEW

**Comment by StevenXL** [comment]
This is a copy-and-paste from #911. Call credit to @meteficha.

File: yesod-auth/Yesod/Auth.hs [hunk 1]
```diff
@@ -421,14 +421,20 @@ authLayoutJson w json = selectRep $ do
 --
 -- @since 1.1.7
 clearCreds :: (MonadHandler m, YesodAuth (HandlerSite m))
-           => Bool -- ^ if HTTP redirect to 'logoutDest' should be done
+           => Bool -- ^ if HTTP redirect to 'logoutDest', if JSON respond with message.
```
Comment (snoyberg) [inline yesod-auth/Yesod/Auth.hs]: It seems a bit strange to me that we conditionally respond with a JSON body like this. What if we _always_ respond with a JSON body if the client accepts JSON responses?

Comment (StevenXL) [reply yesod-auth/Yesod/Auth.hs]: That's a good idea. Will update.

**Comment by StevenXL** [comment]
@snoyberg Not getting much feedback on the failure in CI. Anything I can do?

**Comment by snoyberg** [comment]
No idea, I've never seen it do that before. I'd ignore it, if it remains after a merge to master we can look into it.

**Comment by StevenXL** [comment]
@snoyberg Merged to master. Fingers crossed on the errant error.

**Pull request merged**